### ISSUE TITLE
Seed for _independent_sampler when using SkoptSampler

### ISSUE THREAD

**Issue by sebffischer** [optuna/optuna#2329]
Reproducibility

## Motivation

I wanted to make the hyperparameter-search reproducible by passing "random_state" to the "skopt_kwargs" argument of 
SkoptSampler(). However the results were not reproducible and eventually I solved it via: 

```
import numpy as np 
from optuna.integration.skopt import SkoptSampler 
sampler_agrs = {"random_state": 123, "base_estimator": "GP", "acq_function": "gp_hedge"}
sampler = SkoptSampler(skopt_kwargs=sampler_args)
sampler._independent_sampler._rng = np.random.RandomState(123)
```

## Description

It would be nice to have the argument "seed" for SkoptSampler() so that the seed of the independent sampler is also set.

**Comment by toshihikoyanase** [comment]
Hi @sebffischer 
Thank you for your report.

How about using the `independent_sampler` argument of `SkoptSampler.__init__`?
The following is an example code:

```python
import numpy as np
import optuna

def objective(trial):
    x = trial.suggest_float("x", 0, 1)
    y = trial.suggest_float("y", 0, 1)
    return x ** 2 + y ** 2

seed = 1

sampler = optuna.integration.SkoptSampler(
    independent_sampler=optuna.samplers.RandomSampler(seed),
    skopt_kwargs={"random_state": np.random.RandomState(seed)})
study = optuna.create_study(sampler=sampler)
study.optimize(objective, n_trials=20)
```

I confirmed the reproducibility in the following notebook:
https://colab.research.google.com/drive/1bGmVeWDSWM8i76sneTK0jruXxJSSV2y8?usp=sharing

**Comment by z7ye** [comment]
I also see the same problem.

I was using samplers.TPESampler(seed=42), but would reproduce the same result.

`[I 2021-03-23 03:58:51,446] Using an existing study with name 'hpo_9140df48-d731-4ec4-a224-14d1dd45e52d' instead of creating a new one.
INFO:Optimizing hyperparameters using 150 samples...
[I 2021-03-23 03:58:53,568] Trial 1 finished with value: 0.9598319029897977 and parameters: {'colsample_bytree': 0.7773653241606535, 'learning_rate': 0.3311647820875331, 'max_depth': 5, 'n_estimators': 76, 'subsample': 0.9247636750246644}. Best is trial 1 with value: 0.9598319029897977.
[I 2021-03-23 03:58:54,433] Trial 0 finished with value: 0.9598319029897977 and parameters: {'colsample_bytree': 0.7272064851468021, 'learning_rate': 0.3976900087487727, 'max_depth': 3, 'n_estimators': 125, 'subsample': 0.9021026327502404}. Best is trial 0 with value: 0.9598319029897977.
[I 2021-03-23 03:58:57,030] Trial 2 finished with value: 0.9598319029897977 and parameters: {'colsample_bytree': 0.6506549950471103, 'learning_rate': 0.3373826345044831, 'max_depth': 5, 'n_estimators': 141, 'subsample': 0.9713614645104727}. Best is trial 0 with value: 0.9598319029897977.
[I 2021-03-23 03:58:58,943] Trial 3 finished with value: 0.9598319029897977 and parameters: {'colsample_bytree': 0.6934514051347993, 'learning_rate': 0.3647358163549972, 'max_depth': 3, 'n_estimators': 178, 'subsample': 0.8218172599668634}. Best is trial 0 with value: 0.9598319029897977.
INFO:Finished hyperparemeter search!`


`[I 2021-03-23 03:58:59,234] A new study created with name: hpo_6d7a592b-6d03-47f6-b47f-3fd0e256c116
[I 2021-03-23 03:58:59,271] Using an existing study with name 'hpo_6d7a592b-6d03-47f6-b47f-3fd0e256c116' instead of creating a new one.
INFO:Optimizing hyperparameters using 150 samples...
[I 2021-03-23 03:59:02,499] Trial 0 finished with value: 0.9664818612187034 and parameters: {'colsample_bytree': 0.636065369257361, 'learning_rate': 0.3165617132769272, 'max_depth': 2, 'n_estimators': 125, 'subsample': 0.5965972707694076}. Best is trial 0 with value: 0.9664818612187034.
[I 2021-03-23 03:59:05,258] Trial 2 finished with value: 0.9598319029897977 and parameters: {'colsample_bytree': 0.6915840412565599, 'learning_rate': 0.3417907652354009, 'max_depth': 2, 'n_estimators': 82, 'subsample': 0.9983762938926196}. Best is trial 0 with value: 0.9664818612187034.
[I 2021-03-23 03:59:05,384] Trial 1 finished with value: 0.9598319029897977 and parameters: {'colsample_bytree': 0.7482177791422109, 'learning_rate': 0.307266398968943, 'max_depth': 4, 'n_estimators': 188, 'subsample': 0.9905789587891387}. Best is trial 0 with value: 0.9664818612187034.
[I 2021-03-23 03:59:09,660] Trial 3 finished with value: 0.9532497911445279 and parameters: {'colsample_bytree': 0.7797033706234767, 'learning_rate': 0.38024784935961586, 'max_depth': 1, 'n_estimators': 139, 'subsample': 0.9645349380629027}. Best is trial 0 with value: 0.9664818612187034.
INFO:Finished hyperparemeter search!`

**Comment by sebffischer** [comment]
Even though it is possible with the proposed solution I think it would be more user friendly and intuitive to transfer the seed to the other seeds.
